backends/events/default/default-events.cpp

```diff
@@ -95,8 +95,9 @@ bool DefaultEventManager::pollEvent(Common::Event &event) {
 	event = _eventQueue.pop();
 	bool forwardEvent = true;
 
-	// If the backend has the kFeatureNoQuit, replace Quit event with Return to Launcher
-	if (event.type == Common::EVENT_QUIT && g_system->hasFeature(OSystem::kFeatureNoQuit))
+	// If the backend has the kFeatureNoQuit or the "Return to Launcher at Exit" option is enabled,
+	// replace "Quit" event with "Return to Launcher"
+	if ((event.type == Common::EVENT_QUIT && g_system->hasFeature(OSystem::kFeatureNoQuit)) || ((event.type == Common::EVENT_QUIT && ConfMan.getBool("gui_return_to_launcher_at_exit"))))
```

Comment:
Currently, Command+Q on macOS will no longer quit ScummVM if a game is running when this option is active. IMO, it would be better if exiting a game and quitting ScummVM was two separate events, so that the standard macOS commands for quitting an application would still work as expected.

Reply:
@angstsmurf yes, that's correct - but this is intended behavior. Even closing the ScummVM window on Windows or Linux or closing it with Alt+F4 will only leave the game, but never quit the application in general.

Reply:
That is fine, closing the window or exiting the game should not quit the application, that behaviour has always felt un-Mac-like to me. But the macOS convention is that Command+Q or choosing Quit from the menu (including the dock application menu) should always quit the entire application.

Reply:
Hm, I see - seems like the macOS convention is different here. I must confess that I'm not familiar with macOS at all, so I don't know _anything_ about it. :(

From what I understand, splitting the events in something along the lines of `SOFT_QUIT` and `HARD_QUIT` won't be exactly easy. If this PR introduces breaking regressions on macOS, maybe we should disable it there?

Reply:
@criezy As you proposed, I added an additional check for `g_engine` when calling `ConfMan`.

Reply:
Well, I guess this is a matter of taste, and it would not be the only cross-platform program that will block system shutdown. I don't think I would call it a breaking regression.